src/pgwire/src/protocol.rs

```diff
@@ -1534,13 +1539,8 @@ where
             .map(pgrepr::Type::from)
             .collect::<Vec<pgrepr::Type>>();
 
-        if let State::Ready = next_state {
-            let rows = match decode_copy_text_format(
-                &data,
-                &column_types,
-                &params.delimiter,
-                &params.null,
-            ) {
+        if next_state == State::Ready {
```

Comment:
what's wrong with the Let? that'd allow you to not derive all the stuff on state above.

Reply:
Didn't look as idiomatic to me but reverted to use Let